### ISSUE TITLE
why it is not strict ? 

### ISSUE THREAD

**Issue by paolog22** [stemmlerjs/white-label#15]
can someone please explain why is it not strict thanks

**Comment by chouglesaud** [comment]
@paolog22  what do you mean by ''why is it not strict thanks"

**Comment by Meskat** [comment]
are you talking about the TypeScript setting??

**Comment by paolog22** [comment]
@Meskat yes. when i try making it strict it produces a lot of errors .. @chouglesaud

**Comment by chouglesaud** [comment]
> The strict flag enables a wide range of type checking behavior that results in stronger guarantees of program correctness.

[https://www.typescriptlang.org/tsconfig#strict](https://www.typescriptlang.org/tsconfig#strict)

@paolog22

**Comment by paolog22** [comment]
@chouglesaud i know. but why the code is not ? sorry i am just trying to understand what is behind it .

**Comment by chouglesaud** [comment]
> but why the code is not

can you give an example so I can understand?

`strict: true` will enable all type checking rule like following
  - strictNullChecks
  - alwaysStrict
  - [...](https://www.typescriptlang.org/tsconfig#strict)

for example
if you set `strictNullChecks` to false typescript will just ignore the type checking for variable whether it is null or undefined. on the other hand, if you set to true then it will throw errors.

there is a simple example given in doc's [checkout](https://www.typescriptlang.org/tsconfig#strictNullChecks)

**Comment by chouglesaud** [comment]
> sorry i am just trying to understand what is behind it

everybody here is to learn and understand. I am not professional I just knew a little bit and you @paolog22 help me revising the topic.

**Comment by DadouBordeaux** [comment]
Hi there,

I had a typing problem on my project, here's and sample from the file AfterUserCreated :

```  setupSubscriptions(): void {
    DomainEvents.register(this.onUserCreatedEvent.bind(this), UserCreatedEvent.name);
  }
```
Turn out that when enabling strict mode, the compiler complains with ` Property 'user' is missing in type 'IDomainEvent' but required in type 'UserCreatedEvent'. `
